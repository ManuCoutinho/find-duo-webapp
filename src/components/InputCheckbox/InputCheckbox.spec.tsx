import { screen, render, fireEvent } from '@testing-library/react'
import { InputCheckbox } from '.'
const onChange = jest.fn()

describe('<InputCheckbox/>', () => {
  it('should render InputCheckbox correctly ', () => {
    render(<InputCheckbox onValueChecked={onChange} checked={false} />)
    expect(screen.getByRole('checkbox')).toBeInTheDocument()
  })
  it('should render InputCheckbox unchecked ', () => {
    render(<InputCheckbox onValueChecked={onChange} checked={false} />)
    expect(screen.getByRole('checkbox')).toHaveAttribute(
      'aria-checked',
      'false'
    )
  })
  it('should render InputCheckbox checked ', () => {
    render(<InputCheckbox onValueChecked={onChange} checked={true} />)
    const checkbox = screen.getByRole('checkbox')
    fireEvent.click(checkbox)
    expect(checkbox).toHaveAttribute('aria-checked', 'true')
  })
  it('should match to snapshot ', () => {
    const { container } = render(
      <InputCheckbox onValueChecked={onChange} checked={false} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
