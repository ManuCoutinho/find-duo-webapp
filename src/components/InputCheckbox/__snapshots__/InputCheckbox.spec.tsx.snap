// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputCheckbox/> should match to snapshot  1`] = `
<button
  aria-checked="false"
  class="w-6 h-6 p-1 rounded bg-zinc-900 outline-0 focus-within:ring-1 focus-within:ring-offset-1 focus-within:ring-violet-500 focus-within:outline-violet-600"
  data-state="unchecked"
  id="chat"
  role="checkbox"
  type="button"
  value="on"
/>
`;
